### ISSUE TITLE
Make a gentle intro broken up into bitesize piece per page

### ISSUE THREAD

**Issue by ned14** [ned14/outcome#41]
Notes from review item:

So, I would do something like we have in the documentation for
Boost.Optional.

The initial page is very short. Short description and a really small
example. In fact too small to sell the power of this library:

http://www.boost.org/doc/libs/1_64_0/libs/optional/doc/html/index.html

And then in a number of subsequent pages, we describe feature by feature
with short examples:

http://www.boost.org/doc/libs/1_64_0/libs/optional/doc/html/boost_optional/quick_start.html#boost_optional.quick_start.optional_return_values
http://www.boost.org/doc/libs/1_64_0/libs/optional/doc/html/boost_optional/quick_start/optional_automatic_variables.html
http://www.boost.org/doc/libs/1_64_0/libs/optional/doc/html/boost_optional/quick_start/optional_data_members.html
http://www.boost.org/doc/libs/1_64_0/libs/optional/doc/html/boost_optional/quick_start/bypassing_unnecessary_default_construction.html
http://www.boost.org/doc/libs/1_64_0/libs/optional/doc/html/boost_optional/quick_start/storage_in_containers.html

I would do the same for Outcome:

In the landing page just give this example:

https://github.com/akrzemi1/__sandbox__/blob/master/outcome_intro2.md

It only introduces `outcome::expected` (lot of users will not have heard of
`std::expected`) and the TRY operation to attract attention of more
informed users.

Then in the second page I would provide a number of short capters each with
one annotated example, each introducing one feature:

   - one for using payload wit error_code
   - one for all TRY operations
   - one for interaction with exception_ptr
   - one for catching exceptions into `outcome`
   - one for different ways of accessing the state of `outcome`

If you like this idea, I would volunteer to write up the initial pages.

Regards,
&rzej;

**Comment by ned14** [comment]
Notes from Thomas:

From an empirical study of a non representative group, it seems it is not as literal as you would like it to be.
One thing is that the points I mentioned are not visually distinguishable at first sight, there are three paragraphs with lots of prose.

The first paragraph reads:
"This is the Outcome library. It is a C++ 14 library intended to aid ultra-lightweight error handling in large C++ codebases, providing a more expressive and type safe alternative to integer error codes or enums."

Apart from the trigger word "ultra-lightweight", what the library actually does is at the very end in a long sentence.

The second:
"Unlike alternative implementations, it works perfectly with exceptions and RTTI disabled and is thus suitable for low-latency/games/finance/SG14 users. One could view Outcome as a minimum overhead universal outcome transport mechanism for C++, hence being named "Outcome"."

This sentence starts with "Unlike alternative implementations", I don't even know what to expect from this library, at this point, I really don't care about the alternatives yet, speaking of which, which alternatives? The key feature you wanted to highlight here is probably "works with exceptions and RTTI disabled".

The next two paragraphs follow the same scheme.

So in a nutshell:

The Outcome library implements a set of types to be used for lightweight error transport.
The main features include:
 - Type safe alternative to integer or enum based error codes
 - Works with exceptions and RTTI disabled
 - Suitable for real-time and low latency applications
 - Efficient implementation
 - Rich set of different types to support different use cases

And then you can go on with everything else you want to say, possibly with links pointing into the documentation on where potential users are able to find more about the various topics you just advertised.
(Just a braindump of mine, not claiming this is the ultimate solution.)
